frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckEmailFraudBlockHandler.java

```diff
@@ -76,20 +86,57 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
 
             var emailCheckResult =
                     dynamoEmailCheckResultService.getEmailCheckStore(request.getEmail());
+            var status =
+                    emailCheckResult
+                            .map(EmailCheckResultStore::getStatus)
+                            .orElse(EmailCheckResultStatus.PENDING);
+
+            var checkEmailFraudBlockResponse = createResponse(request.getEmail(), status);
 
-            if (emailCheckResult.isPresent()) {
-                var checkEmailFraudBlockResponse =
-                        new CheckEmailFraudBlockResponse(
-                                request.getEmail(), emailCheckResult.get().getStatus().getValue());
-                return generateApiGatewayProxyResponse(200, checkEmailFraudBlockResponse);
+            if (status.equals(EmailCheckResultStatus.PENDING)) {
+                submitAuditEvent(input, userContext, request);
             }
-            return generateApiGatewayProxyResponse(
-                    200,
-                    new CheckEmailFraudBlockResponse(
-                            request.getEmail(), EmailCheckResultStatus.PENDING.getValue()));
+
+            return generateApiGatewayProxyResponse(200, checkEmailFraudBlockResponse);
         } catch (Json.JsonException e) {
             LOG.error("Unable to serialize check email fraud block response", e);
             throw new RuntimeException(e);
         }
     }
+
+    private CheckEmailFraudBlockResponse createResponse(
+            String email, EmailCheckResultStatus status) {
+        return new CheckEmailFraudBlockResponse(email, status.getValue());
+    }
+
+    private void submitAuditEvent(
+            APIGatewayProxyRequestEvent input,
+            UserContext userContext,
+            CheckEmailFraudBlockRequest request) {
+
+        var clientId =
+                Optional.ofNullable(input.getRequestContext())
```

Comment:
To me, the reason we're having to do this here but not in other handlers warrants some further exploration. I think it might mean we can never get a client id this way, and should instead get it from the user context (see my comment on slack)